script.js

```diff
@@ -1,16 +1,27 @@
 // Write your JavaScript code here!
 
-window.addEventListener("load", function() {
+window.addEventListener("load", function () {
+    let list = document.querySelector("div[id=faultyItems]");
+    list.style.visibility = "hidden";
+    let form = document.querySelector("form");
+    let listedPlanets;
+    let listedPlanetsResponse = myFetch();
 
-   let listedPlanets;
-   // Set listedPlanetsResponse equal to the value returned by calling myFetch()
-   let listedPlanetsResponse;
-   listedPlanetsResponse.then(function (result) {
-       listedPlanets = result;
-       console.log(listedPlanets);
-   }).then(function () {
-       console.log(listedPlanets);
-       // Below this comment call the appropriate helper functions to pick a planet fom the list of planets and add that information to your destination.
-   })
-   
-});
\ No newline at end of file
+    listedPlanetsResponse.then(function (result) {
+        listedPlanets = result;
+        console.log(listedPlanets);
+    }).then(function () {
+        let planet = pickPlanet(listedPlanets);
+        addDestinationInfo(document, planet["name"], planet["diameter"], planet["star"], planet["distance"], planet["moons"], planet["image"]);
+    })
+
+    const button=document.getElementById("formSubmit");
+    button.addEventListener("click",function(e){
+        e.preventDefault();
+        let pilotStatus=document.getElementById("pilotStatus").value;
+        let copilotStatus=document.getElementById("copilotStatus").value;
+        let fuelStatus=document.getElementById("fuelStatus").value;
+        let cargoStatus=document.getElementById("cargoStatus").value;
+        formSubmission(document, pilotStatus, copilotStatus, fuelStatus, cargoStatus)
```

Comment:
```suggestion
        let pilot = document.querySelector("input[name='pilot']");
        let copilot = document.querySelector("input[name='copilot']");
        let fuelLevel = document.querySelector("input[name='fuelLevel']").;
        let cargoMass = document.querySelector("input[name='cargoMass']");
        formSubmission(document, list, pilot.value, copilot.value, fuelLevel.value, cargoMass.value);
```

Reply:
Because the script tags are in the `head` of the HTML. the `getElementById` will not work because, at this point in the code, the elements do not exist. I've added an article below that explains a way to resolve this issue, but instead, you can use `querySelector` instead to get the elements by their name attribute with the attribute selector inside the argument (Ex. `document.querySelector("input[name='pilot']");`, etc.)

This article explains more of ways to resolve it: https://bobbyhadz.com/blog/javascript-cannot-read-property-value-of-null#:~:text=Conclusion%20%23-,The%20%22Cannot%20read%20property%20'value'%20of%20null%22%20error,DOM%20elements%20have%20been%20declared